### DISCUSSION TITLE
Keybinds not working on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by sergeda** [regolith-linux/regolith-desktop#719] [Q&A] [answered]
I installed Regolith 2 on Ubuntu 22.04 as described in the docs but the only key binds that works is super + Enter and sending app to other window. super + shift + Q  and super + alt + Q doesn't work. How can I fix this?

**Comment by kgilmer** [comment]
Hi @sergeda , hmm first thing to check is what got installed.  What does `apt list --installed | grep regolith` return?

**Reply by sergeda** [reply]
Here is what is installed @kgilmer :
```
fonts-nerd-font-bitstreamverasansmono/unknown,now 2.1.0-2-1regolith all [installed,automatic]
i3-gaps-wm/unknown,now 4.20.1-1regolith amd64 [installed,automatic]
i3-next-workspace/unknown,now 1.0.1-1-1regolith amd64 [installed,automatic]
i3xrocks-cpu-usage/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-net-traffic/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-time/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks/unknown,now 1.3.5-1-1regolith amd64 [installed,automatic]
ilia/unknown,now 0.12.14-jammy-1regolith amd64 [installed,automatic]
regolith-compositor-none/unknown,now 1.0.3-1-1regolith amd64 [installed,automatic]
regolith-control-center/unknown,now 1.41.18-1regolith amd64 [installed,automatic]
regolith-default-settings/unknown,now 1.0.2-1regolith amd64 [installed,automatic]
regolith-desktop/unknown,now 4.2.0-1regolith amd64 [installed]
regolith-ftue/unknown,now 2.1.5-1regolith amd64 [installed,automatic]
regolith-i3-base-launchers/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-compositor/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-default-style/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-ftue/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-gaps-partial/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-gnome/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-i3xrocks/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-ilia/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-navigation/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-networkmanager/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-next-workspace/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-resize/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-rofication-ilia/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-session/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-unclutter/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-user-programs/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-workspace-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3xrocks-config/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
regolith-look-default/unknown,now 0.7.1-1regolith amd64 [installed,automatic]
regolith-rofication/unknown,now 1.4.2-1-1regolith amd64 [installed,automatic]
regolith-session-flashback/unknown,now 0.5.0-1regolith amd64 [installed,automatic]
regolith-unclutter-xfixes/unknown,now 1.5-2-1regolith amd64 [installed,automatic]
xrescat/unknown,now 1.2.1-1-1regolith amd64 [installed,automatic]

```

**Comment by kgilmer** [comment]
That all looks good to me @sergeda .  I wonder if another program is intercepting the keybindings.  You can verify it's an install issue by booting into the live session of the Regolith ISO.  If keybindings also do not work there it may be something specific to your setup.

**Comment by Odisej75** [comment]
I am having the same problem. Followed instructions by installing regolith-desktop on Ubuntu 22.04 (all packages upgraded) and shortcuts do not work. I am unable to use just about any keybinding including super + space to start a launcher. Tried to additionally add some packages (rofi et al) but nothing changed. Is there a way to launch the launcher via a command. 

Also changed biding of the super key to ALT but it did not help.

**Comment by Odisej75** [comment]
And the output of the command from the above post:

fonts-nerd-font-bitstreamverasansmono/unknown,now 2.1.0-2-1regolith all [installed,automatic]
i3-gaps-wm/unknown,now 4.20.1-1regolith amd64 [installed,automatic]
i3-next-workspace/unknown,now 1.0.1-1-1regolith amd64 [installed,automatic]
i3xrocks-app-launcher/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-cpu-usage/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-focused-window-name/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-info/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-memory/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks-net-traffic/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
i3xrocks-rofication/unknown,now 5.4.2-1regolith amd64 [installed]
i3xrocks/unknown,now 1.3.5-1-1regolith amd64 [installed,automatic]
ilia/unknown,now 0.12.14-jammy-1regolith amd64 [installed,automatic]
regolith-compositor-none/unknown,now 1.0.3-1-1regolith amd64 [installed,automatic]
regolith-control-center/unknown,now 1.41.18-1regolith amd64 [installed]
regolith-default-settings/unknown,now 1.0.2-1regolith amd64 [installed,automatic]
regolith-desktop/unknown,now 4.2.0-1regolith amd64 [installed]
regolith-ftue/unknown,now 2.1.5-1regolith amd64 [installed,automatic]
regolith-i3-base-launchers/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-compositor/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-default-style/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-gaps-partial/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-i3xrocks/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-navigation/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-networkmanager/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-next-workspace/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-remontoire/unknown,now 3.7.5-1regolith amd64 [installed]
regolith-i3-resize/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-session/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-unclutter/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-user-programs/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3-workspace-config/unknown,now 3.7.5-1regolith amd64 [installed,automatic]
regolith-i3xrocks-config/unknown,now 5.4.2-1regolith amd64 [installed,automatic]
regolith-look-default/unknown,now 0.7.1-1regolith amd64 [installed,automatic]
regolith-rofi-config/unknown,now 1.4.1-1-1regolith amd64 [installed,auto-removable]
regolith-rofication/unknown,now 1.4.2-1-1regolith amd64 [installed,automatic]
regolith-session-flashback/unknown,now 0.5.0-1regolith amd64 [installed,automatic]
regolith-unclutter-xfixes/unknown,now 1.5-2-1regolith amd64 [installed,automatic]
remontoire/unknown,now 1.4.2-1regolith amd64 [installed,automatic]
xrescat/unknown,now 1.2.1-1-1regolith amd64 [installed,automatic]

**Comment by ghost** [comment]
Same problem. All key bindings used to work under Ubuntu 20.04, but are gone after upgrading to 22.04.

**Comment by MAdrianMattocks** [comment]
I'm also having this issue. Worked in 20.04, does not in 22.04. I even tried a fresh install of 22.04 and regolith and still have the issue. Though with the fresh install more of the shortcuts work. I haven't tried them all but the browser one is broken. I checked the 15_launcher file and the shortcut appears to be there. I have also tried reloading the config (that shortcut works).

**Comment by snspinn** [comment]
Just throwing my experience in here. All keybinding working but keybinding with numpad don't.
Definitely not a massive issue. All working so smoothly otherwise.

**Comment by gtheys** [comment]
Have the same problem with the most recent versions. Someone found a solution for it?

**Comment by caioau** [comment, marked as answer]
@sergeda I had the same problem, I end up solving it by installing `regolith-wm-navigation` package which provides the shortcuts on `/usr/share/regolith/common/config.d/30_navigation` so I can navigate to other workspaces with `super + <number>`

Another package missing was `regolith-wm-base-launchers` that makes opening terminal shortcut (super+enter)

cc @gtheys

**Reply by cclamb** [reply]
Unfortunately, this doesn't work for me. I have all the packages you suggest installed. I can't use keybindings for anything - VMWare Fusion, Ubuntu 22.04 LTS, Regolith Desktop 3.2. The VM is configured to pass all keystrokes to the VM, not the host. Raw i3 works just fine - all keybindings work as expected. I also tried to manually set the $mod key in .confg/regolith3/i3/config vai "set $mod Mod4" and that failed as well.

Installed regolith packages:

ayu-theme/unknown,now 0.2.2-1-1regolith-noble arm64 [installed,automatic]
fonts-nerd-font-bitstreamverasansmono/unknown,now 2.1.0-2-1regolith-noble all [installed,automatic]
i3-next-workspace/unknown,now 1.0.3-1regolith-noble arm64 [installed,automatic]
i3-swap-focus/unknown,now 0.4.3-1regolith-noble arm64 [installed,automatic]
i3xrocks-cpu-usage/unknown,now 5.5.1-1regolith-noble arm64 [installed,automatic]
i3xrocks-net-traffic/unknown,now 5.5.1-1regolith-noble arm64 [installed,automatic]
i3xrocks-time/unknown,now 5.5.1-1regolith-noble arm64 [installed,automatic]
i3xrocks/unknown,now 1.3.5-1-1regolith-noble arm64 [installed,automatic]
ilia/unknown,now 0.14.1-1regolith-noble arm64 [installed,automatic]
picom/unknown,now 11.2.2-1regolith-noble arm64 [installed,automatic]
regolith-compositor-picom-glx/unknown,now 1.3.0-1regolith-noble arm64 [installed,automatic]
regolith-control-center/unknown,now 1.46.0-3-1regolith-noble arm64 [installed,automatic]
regolith-default-settings/unknown,now 2.0.4-1regolith-noble arm64 [installed,automatic]
regolith-desktop/unknown,now 4.8.0-1regolith-noble arm64 [installed]
regolith-ftue/unknown,now 2.2.0-1regolith-noble arm64 [installed,automatic]
regolith-i3-compositor/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-control-center-regolith/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-dbus-activation/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-default-style/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-gaps/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-i3xrocks/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-ilia/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-root-config/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-session/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3-unclutter/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-i3xrocks-config/unknown,now 5.5.1-1regolith-noble arm64 [installed,automatic]
regolith-look-default-loader/unknown,now 0.8.2-1regolith-noble arm64 [installed,automatic]
regolith-look-default/unknown,now 0.8.2-1regolith-noble arm64 [installed,automatic]
regolith-look-lascaille/unknown,now 0.9.0-1regolith-noble arm64 [installed]
regolith-rofication/unknown,now 1.5.0-1regolith-noble arm64 [installed,automatic]
regolith-session-common/unknown,now 1.1.12-1regolith-noble arm64 [installed,automatic]
regolith-session-flashback-ext/unknown,now 1.1.12-1regolith-noble arm64 [installed,automatic]
regolith-session-flashback/unknown,now 1.1.12-1regolith-noble arm64 [installed]
regolith-unclutter-xfixes/unknown,now 1.5-2-1regolith-noble arm64 [installed,automatic]
regolith-wm-base-launchers/unknown,now 4.11.6-1regolith-noble arm64 [installed]
regolith-wm-config/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-wm-ftue/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-wm-navigation/unknown,now 4.11.6-1regolith-noble arm64 [installed]
regolith-wm-networkmanager/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-wm-resize/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-wm-rofication-ilia/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-wm-swap-focus/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
regolith-wm-workspace-config/unknown,now 4.11.6-1regolith-noble arm64 [installed,automatic]
xdg-desktop-portal-regolith-x11-config/unknown,now 0.3.5-1regolith-noble arm64 [installed,automatic]
xdg-desktop-portal-regolith/unknown,now 0.3.5-1regolith-noble arm64 [installed,automatic]
xrescat/unknown,now 1.2.1-2-1regolith-noble arm64 [installed,automatic]

**Comment by taylor-shift** [comment]
Doesn't seem to fix it for me...I can't open the launcher with super+space either...weird.

**Comment by darcyrush** [comment]
So I just stumbled across this but in my case I noticed it was actually a graphics card driver issue. My hotkeys seemed unresponsive since all I saw was the status bar (with only the time) and wallpaper - until I moved the mouse around the screen and saw the cursor change.

I opened up TTY3, `tail -f /var/log/syslog`, went back to TTY2 and mashed the enter key, along with the keys for quit and terminal etc and eventually realized I was interacting with the initial setup screen. I managed to choose a timezone without seeing anything on my screen change asides from the cursor changing to busy. Then syslog showed me I was successful with choosing a timezone.

I just happened to notice the following entry in syslog;
```
modprobe ERROR could not insert 'nvidia' key was rejected by service
```
In my case, I had to boot into BIOS, disable `secure boot` and the next login everything worked flawlessly. Before that, I tried using both the dedicated graphics driver and the integrated graphics with both display port and HDMI, but all combinations gave me the same issue. When logging in using Ubuntu rather than Regolith, everything was fine, so I was certain it was hotkey interception, but turns out that wasn't the case.

Maybe this can help someone else in future.
